DXMainClient/DXGUI/Multiplayer/GameInformationPanel.cs

```diff
@@ -29,48 +33,79 @@ public GameInformationPanel(WindowManager windowManager, MapLoader mapLoader)
         private XNALabel lblGameVersion;
         private XNALabel lblHost;
         private XNALabel lblPing;
+        private XNALabel lblPingValue;
         private XNALabel lblPlayers;
         private XNALabel lblSkillLevel;
+
         private XNALabel[] lblPlayerNames;
+        private Texture2D[] pingTextures;
+
+        private GenericHostedGame game = null;
+        private Texture2D mapTexture;
+        private Texture2D pingTexture;
+        private Texture2D noMapPreviewTexture;
+
+        private int leftColumnPositionX = 0;
+        private int rightColumnPositionX = 0;
+        private int topStartingPositionY = 0;
+        private int mapPreviewPositionY = 0;
+        private const int rowHeight = 24;
+        private const int initialPanelHeight = 260;
+        private const int columnWidth = 235;
 
-        private string[] SkillLevelOptions;
+        private string[] skillLevelOptions;
 
         public override void Initialize()
         {
-            ClientRectangle = new Rectangle(0, 0, 235, 288);
+            ClientRectangle = new Rectangle(0, 0, columnWidth * 2, initialPanelHeight);
             BackgroundTexture = AssetLoader.CreateTexture(new Color(0, 0, 0, 255), 1, 1);
             PanelBackgroundDrawMode = PanelBackgroundImageDrawMode.STRETCHED;
 
             lblGameInformation = new XNALabel(WindowManager);
             lblGameInformation.FontIndex = 1;
             lblGameInformation.Text = "GAME INFORMATION".L10N("Client:Main:GameInfo");
 
+            noMapPreviewTexture = AssetLoader.LoadTexture("noMapPreview.png");
```

Comment:
What if it's missing?

Reply:
Mod authors fault? It should be supplied in the theme

Reply:
Couldn't it be optional?

Reply:
I can make it optional if you want

Reply:
Done